compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/collectors/AbstractDiagnosticCollectorVisitor.kt

```diff
@@ -334,6 +336,17 @@ abstract class AbstractDiagnosticCollectorVisitor(
         }
     }
 
+    private inline fun <R> withTypeRefAnnotationContainer(annotationContainer: FirTypeRef, block: () -> R): R {
+        var containingTypeRef = context.annotationContainers.lastOrNull()?.safeAs<FirResolvedTypeRef>()
```

Comment:
Please never user `safeAs`. This function was introduced for cases when `as?` stays in receiver position to avoid redundat parenthesis (`(x as? A)?.foo()` vs `x.safeAs<A>()?.foo()`). But this function is not understandable by compiler, so it won't report `USELESESS_CAST` or `CAST_NEVER_SUCCEED` or smth else if receiver of this call will be changed. Also It has worse IDE support comparing to regular `as?`

I'll fix this one by myself, so it's just note for future

Reply:
Ack